packages/toolpad-app/src/toolpad/AppEditor/NodeMenu.tsx

```diff
@@ -75,6 +88,14 @@ export default function NodeMenu({
           menuProps.onClick?.(event);
         }}
       >
+        {onSettingsNode && (
+          <MenuItem onClick={handleSettingsClick}>
+            <ListItemIcon>
+              <SettingsIcon />
+            </ListItemIcon>
+            <ListItemText>{settingsLabelText}</ListItemText>
+          </MenuItem>
+        )}
```

Comment:
The idea was that a `NodeMenu` could support a generic settings button in the future if `onSettingsNode` was supplied.